pkg/services/database/database_test.go

```diff
@@ -821,3 +821,93 @@ func TestFindPreloadWitLimitAndPage(t *testing.T) {
 		assert.Equal(t, nil, response.GetData())
 	})
 }
+func TestBatch(t *testing.T) {
+	t.Run("should success batch query", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
+		mock.ExpectBegin()
+		mock.ExpectExec(query).WithArgs().WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectCommit()
+
+		database := &database{
+			config:          config.NewDatabaseConfig(),
+			connectionRead:  getMockedConnection(db),
+			connectionWrite: getMockedConnection(db),
+		}
+
+		err = database.RawQuery(query)
+
+		assert.NoError(t, err)
+	})
+	t.Run("should error when exec fails", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
+		mock.ExpectBegin()
+		mock.ExpectExec(query).WithArgs().WillReturnError(errors.New("some error"))
+		mock.ExpectCommit()
+
+		database := &database{
+			config:          config.NewDatabaseConfig(),
+			connectionRead:  getMockedConnection(db),
+			connectionWrite: getMockedConnection(db),
+		}
+
+		err = database.RawQuery(query)
+
+		assert.Error(t, err)
+	})
+	t.Run("should error when begin fails", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
+		mock.ExpectBegin().WillReturnError(errors.New("some error"))
+		mock.ExpectExec(query).WithArgs().WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectCommit()
+
+		database := &database{
+			config:          config.NewDatabaseConfig(),
+			connectionRead:  getMockedConnection(db),
+			connectionWrite: getMockedConnection(db),
+		}
+
+		err = database.RawQuery(query)
+
+		assert.Error(t, err)
+	})
+	t.Run("should error when commit fails", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
+		mock.ExpectBegin()
+		mock.ExpectExec(query).WithArgs().WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectCommit().WillReturnError(errors.New("some error"))
+
+		database := &database{
+			config:          config.NewDatabaseConfig(),
+			connectionRead:  getMockedConnection(db),
+			connectionWrite: getMockedConnection(db),
+		}
+
+		err = database.RawQuery(query)
+
+		assert.Error(t, err)
+	})
+	t.Run("should error when get DB fails", func(t *testing.T) {
+		db := &gorm.DB{
+			Config: &gorm.Config{},
+		}
+
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
```

Comment:
It is a security flaw to pass values in this format see the [GORM documentation](https://gorm.io/docs/sql_builder.html#Raw-SQL) to understand more about it.

Reply:
this is just a sample query that raw query would use, and the function in particular don't use GORM standards or it's functions